x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/engines/engines_list.tsx

```diff
@@ -5,28 +5,89 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useEffect } from 'react';
 
-import { i18n } from '@kbn/i18n';
+import { useActions, useValues } from 'kea';
 
+import { EuiButton, EuiFieldSearch, EuiLink, EuiSpacer, EuiText } from '@elastic/eui';
+
+import { DataPanel } from '../../../shared/data_panel/data_panel';
 import { EnterpriseSearchContentPageTemplate } from '../layout/page_template';
 
+import { EnginesTable } from './components/tables/engines_table';
+import {
+  CREATE_AN_ENGINE_BUTTON_LABEL,
+  ENGINES_DOCUMENTATION,
+  ENGINES_TITLE,
+  ENGINES_TITLE_DESCRIPTION,
+  SEARCH_ENGINE,
+  SEARCH_ENGINE_DESCRIPTION,
+} from './constants';
+import { EnginesListLogic } from './engines_list_logic';
+
 export const EnginesList = () => {
+  const { engines } = useValues(EnginesListLogic);
+
+  const { loadEngines } = useActions(EnginesListLogic);
+
+  useEffect(() => {
+    loadEngines();
+  }, []);
+
   return (
     <EnterpriseSearchContentPageTemplate
-      pageChrome={[
-        i18n.translate('xpack.enterpriseSearch.content.engines.breadcrumb', {
-          defaultMessage: 'Engines',
-        }),
-      ]}
+      pageChrome={[ENGINES_TITLE]}
       pageHeader={{
-        pageTitle: i18n.translate('xpack.enterpriseSearch.content.engines.headerTitle', {
-          defaultMessage: 'Engines',
-        }),
+        pageTitle: ENGINES_TITLE,
+        rightSideItems: [
+          <EuiButton
+            fill
+            iconType="plusInCircle"
+            data-test-subj="appSearchEnginesEngineCreationButton"
+            href={'TODO'}
+          >
+            {' '}
```

Comment:
question: You have several of these spaces across this component. What is the purpose of those spaces?

Reply:
I am Sorry. I think during development, I referred few of the existing code for Create engine button.
Example: [line](https://github.com/saarikabhasi/kibana/blob/saarikabhasi-engine-list-page-1/x-pack/plugins/enterprise_search/public/applications/app_search/components/curations/curation/automated_curation.tsx#L72)

And I missed to remove those spaces before raising PR.  Will correct.